coala_quickstart/info_extractors/GruntfileInfoExtractor.py

```diff
@@ -0,0 +1,275 @@
+
+        return results
+
+    def extract_lint_subtasks(self, parsed_file):
+        """
+        Extract the lint subtasks from the parsed JS file.
+        Looks for identifiers like:
+        ``grunt.registerTask( 'lint', [ 'csslint', 'jshint' ] );``
+
+        :param parsed_file:
+            An instance of PyJsParser().parse
+        :return:
+            A list of lint subtasks
+        """
+        # task names to match
+        keys_to_match = ['lint']
+
+        # Serch for grunt.registerTask() identifiers
+        search_results = search_object_recursively(parsed_file, "callee", {
+            "computed": False,
+            "type": "MemberExpression",
+            "property": {
+                "name": "registerTask",
+                "type": "Identifier"
+                },
+            "object": {
+                "name": "grunt",
+                "type": "Identifier"
+                }
+            })
+
+        lint_subtasks = []
+        is_lint = False
+
+        for res in search_results:
+            arguments = res["object"].get('arguments')
+            # check if the task is a lint sub-task
+            if arguments:
+                for arg in arguments:
+                    if arg.get('value') in keys_to_match:
+                        is_lint = True
+                        break
+            if is_lint:
+                for arg in arguments:
+                    if ("elements" in arg.keys() and
+                            arg["type"] == "ArrayExpression"):
+                        lint_subtasks += [
+                            elem["value"].split(':')[0]
+                            for elem in arg["elements"]
+                        ]
+                        is_lint = False
+        return lint_subtasks
+
+    def get_configurations(self, parsed_file, tasks):
+        """
+        Extract the configurations from the PyJsParser instance
+        of a Gruntfile and return configuration data(if any) of
+        the task provided. Looks for patterns like
+        {
+            grunt.initConfig(
+                {
+                    some configuration objects
+                }
+        }
+        :param parsed_file:
+            An instance of PyJsParser().parse
+        :param tasks:
+            list of task names for which configurations are
+            to be extracted.
+        :return:
+            A list of configuration dicts
+        """
+        result = {}
+        search_results = search_object_recursively(
+            parsed_file, "callee",
+            {
+                "computed": False,
+                "type": "MemberExpression",
+                "property": {
+                    "name": "initConfig",
+                    "type": "Identifier"
+                },
+                "object": {
+                    "name": "grunt",
+                    "type": "Identifier"
+                }
+            })
+
+        try:
+            for arg in search_results[-1]["object"]["arguments"]:
```

Comment:
Will there be more than one result? If so will it only use the last one? Or will it use the first one, then override existing fields in the second call, third call, ...